lite2/client.go

```diff
@@ -767,24 +768,22 @@ func (c *Client) bisection(
 			c.trustLevel)
 		switch err.(type) {
 		case nil:
+			if interimHeader.Height == newHeader.Height {
+				return nil
+			}
+
 			// Update the lower bound to the previous upper bound
-			trustedHeader = interimHeader
-			trustedNextVals, err = c.validatorSetFromPrimary(interimHeader.Height + 1)
+			interimVals, err = c.validatorSetFromPrimary(interimHeader.Height + 1)
 			if err != nil {
 				return err
 			}
-			if !bytes.Equal(trustedHeader.NextValidatorsHash, trustedNextVals.Hash()) {
+			if !bytes.Equal(interimHeader.NextValidatorsHash, interimVals.Hash()) {
 				return errors.Errorf("expected next validator's hash %X, but got %X (height #%d)",
-					trustedHeader.NextValidatorsHash,
-					trustedNextVals.Hash(),
-					trustedHeader.Height)
+					interimHeader.NextValidatorsHash,
+					interimVals.Hash(),
+					interimHeader.Height)
 			}
-
-			err = c.updateTrustedHeaderAndNextVals(trustedHeader, trustedNextVals)
-			if err != nil {
-				return err
-			}
-
+			trustedNextVals, trustedHeader = interimVals, interimHeader
```

Comment:
same thing - try to be consistent

Reply:
Can you further explain, what am I not being consistent with

Reply:
In the above sequence func we have `trustedHeader, trustedNextVals =` while here we have `trustedNextVals, trustedHeader =`